iree/compiler/Dialect/HAL/Conversion/TypeConverter.cpp

```diff
@@ -33,9 +33,14 @@ HALTypeConverter::HALTypeConverter(
     results.push_back(type);
     return success();
   });
-  addConversion([](TensorType type) {
-    // TODO(benvanik): composite-type conversion (buffer + dynamic dims).
-    return IREE::HAL::BufferType::get(type.getContext());
+  addConversion([](TensorType type) -> Optional<Type> {
+    // HAL only should be concerned with numeric values.
+    if (HALTypeConverter::ConvertToHalBuffer(type)) {
```

Comment:
just say `type.getElementType().isIntOrFloat()`. You don't need the helper.

Reply:
Oh, I see this is used various places. Maybe keep it but call it ShouldConvertToHalBuffer? The current name doesn't make much sense to me given what the function does.

Reply:
Changed to ShouldConvertToHalBuffer.